fix: order item imageURLs

tedraykov opened the pull request:
Impact: **minor**
Type: **bugfix**

## Issue
The image URLs for order items are not being fetched and are not served in the request payload. The same bug was present in the `carts` plugin and was fixed by this PR https://github.com/reactioncommerce/api-plugin-carts/commit/7c2a153572b1dbcefc5faf0ed9a9c65a46b9ed45

## Solution
As the order item images are needed in the email templates, for example, they need to be added in the `xformOrderItems`.

## Testing
Response before the fix:
```
"items": {
  "nodes": [
    {
      "title": "Primary 1 Product",
      "imageURLs": null
    }
  ]
}
```

Response after the fix:
```
"items": {
  "nodes": [
    {
      "title": "Primary 1 Product",
      "imageURLs": {
        "large": "http://localhost:3000/assets/files/Media/BknHFoW8sypPdGXyB/large/banner.jpg",
        "medium": "http://localhost:3000/assets/files/Media/BknHFoW8sypPdGXyB/medium/banner.jpg",
        "original": "http://localhost:3000/assets/files/Media/BknHFoW8sypPdGXyB/image/banner.jpg",
        "small": "http://localhost:3000/assets/files/Media/BknHFoW8sypPdGXyB/small/banner.png",
        "thumbnail": "http://localhost:3000/assets/files/Media/BknHFoW8sypPdGXyB/thumbnail/banner.png"
      }
    }
  ]
}
```

tedraykov committed:
fix: order item imageURLs

Signed-off-by: tedraykov <tedraykov@gmail.com>

brent-hoover commented:
@tedraykov Tests are failing here

https://app.circleci.com/pipelines/github/reactioncommerce/api-plugin-orders/116/workflows/56389f62-47bd-473a-932a-3d0389a69234/jobs/322

tedraykov committed:
test: update data for email test

Await imageURLs result for all items in `xformOrderItems.js`
Added the imageURLs and backwards compatible `productImage` and
`variantImage` attributes of order items.
Removed legacy mocking steps as they are no longer used in tests.

Signed-off-by: tedraykov <tedraykov@gmail.com>

tedraykov commented:
@zenweasel The failing test is addressed. Also removed some deprecated mocks in the test.

rc-publisher commented:
:tada: This PR is included in version 1.4.4 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@reactioncommerce/api-plugin-orders/v/1.4.4)
- [GitHub release](https://github.com/reactioncommerce/api-plugin-orders/releases/tag/v1.4.4)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
